core/src/main/java/org/elasticsearch/bootstrap/BootstrapCheck.java

```diff
@@ -59,13 +60,14 @@ private BootstrapCheck() {
      * checks
      *
      * @param settings              the current node settings
+     * @param dataDirectoryNumber   the integer number of the directories who's locks under the data directory we were able to take
```

Comment:
I've renamed it to `dataDirectoryIndex` because it feels like an array index because it is 0 based.